src/ipc/zone/event_scene.rs

```diff
@@ -12,11 +12,11 @@ pub struct EventScene {
     #[brw(pad_before = 2)] // FIXME: um, i don't think this is empty!!
     pub scene_flags: u32,
     pub unk1: u32,
-    pub unk2: u8,
-    #[brw(pad_before = 3)]
-    pub unk3: u32,
-    pub unk4: u32,
-    pub unk5: u32,
+    pub params_count: u8,
+    #[brw(pad_before = 3, pad_after = 4)]
+    // Extra padding seems needed after or the client will seemingly softlock even with 2 params, possibly used for alignment?
+    #[br(count=params_count)]
+    pub params: Vec<u32>,
```

Comment:
So the rest of the changes are correct, but this isn't isn't. No packet in the game is dynamic, each is a fixed size so this one probably only supports maybe 2 or 4 parameters (Sapphire might confirm this.)

Instead of defining a `Vec<u32>`, it would be a `[u32; 2]` or something, yes I know the `params_count` makes this somewhat confusing but it's always _less_ than what the packet actually supports. (The rest of the packet is considered invalid data.)

In the future it would be nice to make this generic, so we define each like `EventScene<2>` (for 2 params), `EventScene<4>` (4 params) or whatever but just supporting the 2 for right now is good.